rbac_api_with_domains.go

```diff
@@ -144,17 +144,35 @@ func (e *Enforcer) DeleteAllUsersByDomain(domain string) (bool, error) {
 	return true, nil
 }
 
-// DeleteDomains would delete all associated users and roles.
+// DeleteDomains would delete all associated policies.
 // It would delete all domains if parameter is not provided.
 func (e *Enforcer) DeleteDomains(domains ...string) (bool, error) {
 	if len(domains) == 0 {
-		e.ClearPolicy()
+		allDomains, _ := e.GetAllDomains()
+		for _, domain := range allDomains {
+			if _, err := e.DeleteDomains(domain); err != nil {
+				return false, err
+			}
+		}
```

Comment:
The logic here can be combined with the following, just use `e. GetAllDomains `at the beginning. 
such as
```go
allDomains := domains
if len(domains) == 0 {
    allDomains, _ := e.GetAllDomains()
}
ffor _, domain := range allDomains {
```